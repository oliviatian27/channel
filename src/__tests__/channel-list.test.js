import React from 'react';
import { shallow } from 'enzyme';

import ChannelList from '../components/channel-list';


describe('ChannelList tests', () => {

  it('renders Channel-items', () => {
    const list = {data:[{'title':'one'}, {'title':'two'}, {'title':'three'}],date:"2018-11-01"};
    const wrapper = shallow(<ChannelList list={list} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-list')).toBeDefined();
    expect(wrapper.find('.channel-list')).toMatchSnapshot();
  });

  it('renders a ChannelDate component ', () => {
    const list = {data:[{'title':'one'}, {'title':'two'}, {'title':'three'}],date:"2018-11-01"};
    const wrapper = shallow(<ChannelList list={list} />);

    // Check if an element in the Component exists
    expect(wrapper.find('ChannelDate').length).toEqual(1);
  });

  it('renders a list of Channel components ', () => {
    const list = {data:[{'title':'one'}, {'title':'two'}, {'title':'three'}],date:"2018-11-01"};
    const wrapper = shallow(<ChannelList list={list} />);

    // Check if an element in the Component exists
    expect(wrapper.find('Channel').length).toEqual(list.data.length);
  });

});
