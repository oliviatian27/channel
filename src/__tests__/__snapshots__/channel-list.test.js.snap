// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelList tests renders Channel-items 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelList
      isRequired={[Function]}
      list={
        Object {
          "data": Array [
            Object {
              "title": "one",
            },
            Object {
              "title": "two",
            },
            Object {
              "title": "three",
            },
          ],
          "date": "2018-11-01",
        }
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ChannelDate
            date="2018-11-01"
          />,
          Array [
            <Channel
              data={
                Object {
                  "title": "one",
                }
              }
            />,
            <Channel
              data={
                Object {
                  "title": "two",
                }
              }
            />,
            <Channel
              data={
                Object {
                  "title": "three",
                }
              }
            />,
          ],
        ],
        "className": "channel-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "date": "2018-11-01",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "one",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "two",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "three",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ChannelDate
              date="2018-11-01"
            />,
            Array [
              <Channel
                data={
                  Object {
                    "title": "one",
                  }
                }
              />,
              <Channel
                data={
                  Object {
                    "title": "two",
                  }
                }
              />,
              <Channel
                data={
                  Object {
                    "title": "three",
                  }
                }
              />,
            ],
          ],
          "className": "channel-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "date": "2018-11-01",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "title": "one",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "title": "two",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "title": "three",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ChannelDate
          date="2018-11-01"
        />,
        Array [
          <Channel
            data={
              Object {
                "title": "one",
              }
            }
          />,
          <Channel
            data={
              Object {
                "title": "two",
              }
            }
          />,
          <Channel
            data={
              Object {
                "title": "three",
              }
            }
          />,
        ],
      ],
      "className": "channel-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "date": "2018-11-01",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "title": "one",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "title": "two",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "title": "three",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ChannelDate
            date="2018-11-01"
          />,
          Array [
            <Channel
              data={
                Object {
                  "title": "one",
                }
              }
            />,
            <Channel
              data={
                Object {
                  "title": "two",
                }
              }
            />,
            <Channel
              data={
                Object {
                  "title": "three",
                }
              }
            />,
          ],
        ],
        "className": "channel-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "date": "2018-11-01",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "one",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "two",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "three",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ChannelDate
            date="2018-11-01"
          />,
          Array [
            <Channel
              data={
                Object {
                  "title": "one",
                }
              }
            />,
            <Channel
              data={
                Object {
                  "title": "two",
                }
              }
            />,
            <Channel
              data={
                Object {
                  "title": "three",
                }
              }
            />,
          ],
        ],
        "className": "channel-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "date": "2018-11-01",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "one",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "two",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "title": "three",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
}
`;
